go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -457,38 +457,43 @@ func (c *Controller) syncNodeInfos(nodeInfos []nodeInfo) {
 	}
 
 	// Compute the nodeIP template values.
-	c.nodeIPv4Template = makeTemplate(makeLBNodeIPTemplateName(v1.IPv4Protocol))
-	c.nodeIPv6Template = makeTemplate(makeLBNodeIPTemplateName(v1.IPv6Protocol))
+	c.nodeIPv4Templates = NewNodeIPsTemplates(v1.IPv4Protocol)
+	c.nodeIPv6Templates = NewNodeIPsTemplates(v1.IPv6Protocol)
 
-	for _, node := range c.nodeInfos {
-		if node.chassisID == "" {
+	for _, nodeInfo := range c.nodeInfos {
+		if nodeInfo.chassisID == "" {
 			continue
 		}
-		// Services are currently supported only on the node's first IP.
-		// Extract that one and populate the node's IP template value.
+
 		if globalconfig.IPv4Mode {
-			if ipv4, err := util.MatchFirstIPFamily(false, node.l3gatewayAddresses); err == nil {
-				c.nodeIPv4Template.Value[node.chassisID] = ipv4.String()
+			ips, err := util.MatchIPFamily(false, nodeInfo.hostAddresses)
+			if err != nil {
+				klog.Warningf("Error while searching for IPv4 host addresses: %v", err)
```

Comment:
we should probably include the nodeInfo.name in the warning message to make it easier to map it to the node